server/src/main/java/org/opensearch/search/fetch/subphase/highlight/AbstractHighlighterBuilder.java

```diff
@@ -130,7 +130,7 @@
 
     protected Boolean requireFieldMatch;
 
-    protected int maxAnalyzerOffset = -1;
+    protected Integer maxAnalyzerOffset = null;
```

Comment:
Please, do not forget to update this implementations with the new field:

`AbstractHighlighterBuilder(StreamInput in)`
`public final void writeTo(StreamOutput out) throws IOException`

Reply:
Finally working.